core/pom.xml

```diff
@@ -179,6 +185,14 @@
                <exclude>schema/**</exclude>
             </excludes>
          </resource>
+         <resource>
+            <directory>${project.build.outputDirectory}/schema</directory>
+            <filtering>false</filtering>
+            <includes>
+               <include>generated/*.proto</include>
+            </includes>
+            <targetPath>schema</targetPath>
```

Comment:
I tried todo this, however I run into issues with checkstyle now checking the generated `Log.i18n.properties` file and failing. I tried reading the entire build dir and excluding paths with no luck. So I figured narrowing the search to just `schema/` dir and using `targetPath` was acceptable.

Reply:
@ryanemerson I didn't know we had checkstyle rules for properties files, but why would checkstyle ever check files in `target/classes`? For ISPN-10137 I'm generating source files with plenty of unused imports in `target/generated-test-sources`, and checkstyle never complained.

Reply:
To answer my own question, it's probably because I never added `target/generated-test-sources` as an explicit source directory in Maven, the compiler plugin added it, and that's why Checkstyle didn't include it. So perhaps the solution in your case is to remove the `<resource>` tag, because everything in `target/classes` is included in the jar anyway.

Oh, and perhaps we should move the Checkstyle execution before compilation.

Reply:
@danberindei For some reason with core the generated proto files are not included in the jar without my `<resource>` definition :confused:

Reply:
Sorry, I have a really short memory: I had to add the `META-INF/services` directory to the `<Include-Resource>` bundle instructions configuration myself when working on ISPN-10137 :) It's probably because the regular jar plugin doesn't care what's in `target/classes`, but `maven-bundler-plugin` is more anal about including only .class files and the resources that `maven-resource-plugin` saw in the `resource` phase (which is before `compile`)

Since we already use the `<Include-Resource>` hack for other resources, I'd use it for proto schemas as well, and it should take care of checkstyle.

Reply:
I've also applied this to `infinispan-lucene-directory`, `infinispan-remote-query-server` and `infinispan-server-memcached`.